pkg/workerapi/workerserver/httpserver.go

```diff
@@ -0,0 +1,158 @@
+package workerserver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gin-contrib/cors"
+	"github.com/gin-gonic/gin"
+	"github.com/iver-wharf/wharf-core/pkg/ginutil"
+)
+
+type module interface {
+	register(g *gin.RouterGroup)
+}
+
+type httpServer struct {
+	bindAddress  string
+	srv          *http.Server
+	workerServer *workerHTTPServer
+	isRunning    bool
+	serveMutex   *sync.Mutex
+}
+
+// NewHTTPServer creates a new HTTP server that can be started by calling Serve.
+func NewHTTPServer(
+	bindAddress string,
+	buildStepLister BuildStepLister,
+	artifactLister ArtifactLister,
+	artifactDownloader ArtifactDownloader) Server {
+	return &httpServer{
+		bindAddress:  bindAddress,
+		workerServer: newWorkerHTTPServer(buildStepLister, artifactLister, artifactDownloader),
```

Comment:
Is there a way currently to get the functionality of `BuildStepLister` (now `StepLister`) from `resultstore.Store`?

Reason for the split up is because I'm not entirely clear on which type is to be used for the separate things, if `resultstore.Store` can do all of it (except for artifact atm), then I can't seem to find it.

Reply:
This should look a lot nicer now, with the merging as well.
HTTP has been stripped down to only downloading, added artifact related stuff to `resultstore.Store`.